src/Microsoft.ML.Core/Data/MetadataUtils.cs

```diff
@@ -312,7 +312,6 @@ public static bool HasSlotNames(this ISchema schema, int col, int vectorSize)
         public static void GetSlotNames(RoleMappedSchema schema, RoleMappedSchema.ColumnRole role, int vectorSize, ref VBuffer<DvText> slotNames)
         {
             Contracts.CheckValueOrNull(schema);
-            Contracts.CheckValue(role.Value, nameof(role));
```

Comment:
>Contracts.CheckValue(role.Value, nameof(role)); [](start = 12, length = 47)

you removing it because ColumnRole has only one constuctor in which we invoke Contracts.CheckNonEmpty, so Value always gonna have Value, right?

Reply:
Right. In the issue #445 when I said "Also a fair amount of code exists in the class to detect conditions that cannot possibly happen." this is partially what I am referring to. Indeed I think it's the only real reason to have a separate struct instead of just using `string` directly, so it's a little weird that we spent so much effort checking and rechecking.